pybamm/models/full_battery_models/lithium_ion/basic_dfn_half_cell.py

```diff
@@ -0,0 +1,474 @@
+#
+# Basic Doyle-Fuller-Newman (DFN) Half Cell Model
+#
+import pybamm
+from .base_lithium_ion_model import BaseModel
+
+
+class BasicDFNHalfCell(BaseModel):
+    """Doyle-Fuller-Newman (DFN) model of a lithium-ion battery with lithium counter
+    electrode, adapted from [2]_.
+
+    This class differs from the :class:`pybamm.lithium_ion.BasicDFN` model class in
+    that it is for a cell with a lithium counter electrode (half cell). This is a
+    feature under development (for example, it cannot be used with the Simulation class
+    for the moment) and in the future it will be incorporated as a standard model with
+    the full functionality.
+
+    Parameters
+    ----------
+    name : str, optional
+        The name of the model.
+    options : dict
+        A dictionary of options to be passed to the model. For the half cell it should
+        include which is the working electrode.
+
+    References
+    ----------
+    .. [2] M Doyle, TF Fuller and JS Nwman. “Modeling of Galvanostatic Charge and
+        Discharge of the Lithium/Polymer/Insertion Cell”. Journal of The
+        Electrochemical Society, 140(6):1526-1533, 1993
+
+    **Extends:** :class:`pybamm.lithium_ion.BaseModel`
+    """
+
+    def __init__(
+        self,
+        name="Doyle-Fuller-Newman half cell model",
+        options={"working electrode": None},
```

Comment:
Safer to do `options=None` and then in the first line of the function
```
options = options or {"working electrode": None}
```
which is the same as
```
if options is None:
     options = {"working electrode": None}
```

Reply:
Done